### ISSUE TITLE
mCRL2xi can be crashed with an incorrect specification

### ISSUE THREAD

**Issue by gjveltink** [mCRL2org/mCRL2#1789]
When trying to rewrite 'makeList(-1)' using the following specification within mCRL2xi the application crashes.

--- 

```
map
	makeList: Int -> List(Int);

var
	n: Int;

eqn
	% makeList(n) = if(n <= 0, [], makeList(n - 1) <| n);
	makeList(n) = if(n == 0, [], makeList(n - 1) <| n); % rewrite of 'makeList(-1)' crashes mCRL2xi 
```

---

[Problem.mcrl2.txt](https://github.com/user-attachments/files/17489637/Problem.mcrl2.txt)

**Comment by gjveltink** [comment]
Yes, I understand that mCRL2xi can crash with this specification, but I seem to remember, that it did not do so in the past. However, maybe I'm confusing this with the behaviour of mcrl2ide.

**Comment by gjveltink** [comment]
sorry, closed this inadvertently.

**Comment by mlaveaux** [comment]
One discussion we had during the mCRL2 meeting is if it would be useful to remove mcrl2xi in favor of just having the mcrl2ide instead of adapting mcrl2xi for the next release. Would this be something that could work for you, or are there features in mcrl2ide that are missing for this to be a good alternative.

**Comment by gjveltink** [comment]
I have to admit, that up to now I was not aware of the fact that mcrl2ide now can rewrite data terms. I will try this out in class next week! The only thing that might be missing is the setting of (some of) the different options, that can be set in mcrl2xi.

**Comment by gjveltink** [comment]
OK, I have given it a try in mcrl2ide and have observed that, like in mcrl2xi before, I get a syntax error when the "init" clause is missing. :-(
